clang/lib/Parse/ParseDecl.cpp

```diff
@@ -6750,12 +6750,24 @@ void Parser::ParseDeclaratorInternal(Declarator &D,
     // Is a reference
     DeclSpec DS(AttrFactory);
 
-    // Complain about rvalue references in C++03, but then go on and build
-    // the declarator.
-    if (Kind == tok::ampamp)
-      Diag(Loc, getLangOpts().CPlusPlus11 ?
-           diag::warn_cxx98_compat_rvalue_reference :
-           diag::ext_rvalue_reference);
+    // Complain about:
+    // - rvalue references in C++03
+    // - `^T &&` parsing ambiguity of compound expression with reflection in C++2x
```

Comment:
Hmm now that I think of it, we probably need it for `&` as well -

```cpp
constexpr auto r = ^int;
constexpr bool b = (r != ^char & true);
```

Would you mind diagnosing this as well?

Reply:
Sure, will push later today. 
Need to figure out how to test several parsing errors in 1 test file. 
In compiler explorer I can see parsing error so I expect that in tests I could configure it somehow as well

Reply:
Fixed. Changed code and test a little bit. 

Tested manually as well, got expected warning:
```cpp
main.cpp:10:28: warning: the compound condition may be misinterpreted due to 'T &' type parsing logic. did you mean '... (^T) & ...'? [-Wreflect-parsing-ambiguity]
   10 |   if (reflValue != ^MyEnum & true) {
      |                            ^

```